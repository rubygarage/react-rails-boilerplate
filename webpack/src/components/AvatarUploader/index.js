import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';

import styles from './styles.css';

function AvatarUploader({
  input, name, multiple, handleRemoveSelectedFile, fileDropHandler,
}) {
  const files = input.value;
  return (
    <div>
      <Dropzone
        multiple={multiple}
        name={name}
        accept="image/jpeg, image/png"
        onDrop={fileDropHandler}
      >
        { files && Array.isArray(files)
          ?
            files.map(file => (
              <div className={styles['avatar-preview-wrap']} key={file.name}>
                <img src={file.preview} alt="Uploaded avatar" className={styles['avatar-uploader-thumb']} />
                <button
                  type="button"
                  className={styles['remove-avatar-preview-button']}
                  onClick={handleRemoveSelectedFile}
                >
                  x
                </button>
              </div>
            ))
          :
            <div>Try dropping some files here, or click to select files to upload.</div>
        }
      </Dropzone>
    </div>
  );
}

AvatarUploader.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
    ]),
    onChange: PropTypes.func,
  }),
  handleRemoveSelectedFile: PropTypes.func.isRequired,
  fileDropHandler: PropTypes.func,
  multiple: PropTypes.bool,
  name: PropTypes.string,
};

export default AvatarUploader;
