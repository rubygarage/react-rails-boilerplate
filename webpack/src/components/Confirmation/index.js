import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

function Confirmation() {
  return (
    <main>
      <div className="jumbotron">
        <h4> <FormattedMessage id="email_confirmation.success" /> </h4>

        <Link to="/sign_in">
          <p>
            <FormattedMessage id="email_confirmation.login" />
          </p>
        </Link>
      </div>
    </main>
  );
}

export default Confirmation;
