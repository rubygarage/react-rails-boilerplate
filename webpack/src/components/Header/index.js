import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

function Header({ currentUser, currentUserProfileLink, handleSignOut }) {
  return (
    <nav className="navbar primary-color">
      <form className="form-inline justify-content-end">
        {currentUser.id ? (
          <div>
            <Link className="btn btn-primary mr-2" to={currentUserProfileLink}>
              <FormattedMessage id="header.my_profile" />
            </Link>
            <button className="btn btn-primary" onClick={handleSignOut} >
              <FormattedMessage id="header.logout" />
            </button>
          </div>
        ) : (
          <Link className="btn btn-primary" to="/sign_in">
            <FormattedMessage id="header.signin" />
          </Link>
        )}
      </form>
    </nav>
  );
}

Header.propTypes = {
  currentUser: PropTypes.shape({
    id: PropTypes.string,
    username: PropTypes.string,
    email: PropTypes.string,
  }),
  currentUserProfileLink: PropTypes.string,
  handleSignOut: PropTypes.func.isRequired,
};

export default Header;
