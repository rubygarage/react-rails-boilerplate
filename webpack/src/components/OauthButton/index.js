import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/ui/Button';
import Icon from 'components/ui/Icon';

function OauthButton({ color, type, onClick }) {
  return (
    <Button color={color} size="md" className="mb-20" onClick={onClick}>
      <Icon type={type} className="social-icon" />
    </Button>
  );
}

OauthButton.propTypes = {
  color: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
};

export default OauthButton;
