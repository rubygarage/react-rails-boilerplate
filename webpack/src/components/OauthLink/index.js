import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'components/ui/Icon';
import Button from 'components/ui/Button';

function OauthLink({ type, onClick }) {
  return (
    <Button className="social-round-icon" onClick={onClick}>
      <Icon type={type} />
    </Button>
  );
}

OauthLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
};

export default OauthLink;
