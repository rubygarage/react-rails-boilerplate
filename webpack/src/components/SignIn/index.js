import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedMessage, intlShape } from 'react-intl';
import { Field } from 'redux-form';
import { Link } from 'react-router-dom';
import ErrorMessage from 'components/ui/ErrorMessage';

import styles from './styles.css';

export function SignIn({
  handleSubmit,
  pristine,
  submitHandler,
  submitting,
  intl: { formatMessage },
}) {
  return (
    <div className="container">
      <div className="row">

        <div className="col-md-6 offset-md-3">
          <div className={styles['register-wrap']}>
            <div className="card">
              <div className="card-header text-center">
                <h2> <FormattedMessage id="sign_in.login" /> </h2>
              </div>
              <div className="card-block">
                <form onSubmit={handleSubmit(submitHandler)}>
                  <ErrorMessage name="username" />

                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="username"
                      placeholder={formatMessage({ id: 'sign_up.username' })}
                      aria-label={formatMessage({ id: 'sign_up.username' })}
                    />
                  </div>

                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password"
                      placeholder={formatMessage({ id: 'sign_up.password' })}
                      aria-label={formatMessage({ id: 'sign_up.password' })}
                    />
                  </div>

                  <div>
                    <Field id="remember" name="remember" component="input" className="mr-2" type="checkbox" />
                    <label htmlFor="remember">
                      <FormattedMessage id="sign_in.remember_me" />
                    </label>
                  </div>

                  <button className="btn btn-primary mb-3" type="submit" disabled={pristine || submitting}>
                    <FormattedMessage id="sign_in.login" />
                  </button>

                </form>
                <p>
                  <FormattedMessage id="sign_in.not_a_member_yet" />
                  &nbsp;
                  <Link to="/sign_up">
                    <FormattedMessage id="sign_in.register" />
                  </Link>
                </p>

                <p>
                  <Link to="/reset_password">
                    <FormattedMessage id="sign_in.forgot_password" />
                  </Link>
                </p>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  );
}

SignIn.propTypes = {
  intl: intlShape,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitHandler: PropTypes.func.isRequired,
  submitting: PropTypes.bool,
};

export default injectIntl(SignIn);
