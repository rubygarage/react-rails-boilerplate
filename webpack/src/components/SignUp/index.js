import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, intlShape } from 'react-intl';
import { Link } from 'react-router-dom';
import { Field } from 'redux-form';
import ErrorMessage from 'components/ui/ErrorMessage';
import Oauth from 'containers/Oauth';

import styles from './styles.css';

function SignUp({
  handleSubmit, submitHandler, pristine, submitting, intl: { formatMessage },
}) {
  return (
    <div className="container">
      <div className="row">

        <div className="col-md-6 offset-md-3">
          <div className={styles['register-wrap']}>
            <div className="card">
              <div className="card-header text-center">
                <h2> <FormattedMessage id="sign_up.register" /> </h2>
              </div>
              <div className="text-center">
                <Oauth type="icon-fb" provider="facebook" button />
              </div>
              <div className="card-block">
                <form onSubmit={handleSubmit(submitHandler)}>
                  <ErrorMessage name="username" />
                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="username"
                      placeholder={formatMessage({ id: 'sign_up.username' })}
                      aria-label={formatMessage({ id: 'sign_up.username' })}
                    />
                  </div>

                  <ErrorMessage name="email" />
                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="email"
                      placeholder={formatMessage({ id: 'sign_up.email' })}
                      aria-label={formatMessage({ id: 'sign_up.email' })}
                    />
                  </div>

                  <ErrorMessage name="password" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password"
                      placeholder={formatMessage({ id: 'sign_up.password' })}
                      aria-label={formatMessage({ id: 'sign_up.password' })}
                    />
                  </div>

                  <ErrorMessage name="password_confirmation" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password_confirmation"
                      placeholder={formatMessage({ id: 'sign_up.confirm_password' })}
                      aria-label={formatMessage({ id: 'sign_up.confirm_password' })}
                    />
                  </div>

                  <button className="btn btn-primary mb-3" type="submit" disabled={pristine || submitting}>
                    <FormattedMessage id="sign_up.register" />
                  </button>
                </form>

                <p>
                  <FormattedMessage id="sign_up.already_member" />
                  &nbsp;
                  <Link to="/sign_in">
                    <FormattedMessage id="sign_up.login" />
                  </Link>
                </p>

              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  );
}

SignUp.propTypes = {
  intl: intlShape,
  handleSubmit: PropTypes.func.isRequired,
  submitHandler: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
};

export default SignUp;
