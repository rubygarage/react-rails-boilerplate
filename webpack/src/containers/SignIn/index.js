import React, { Component } from 'react'
import { reduxForm } from 'redux-form'
import submit from './submit'

import SignInComponent from 'components/SignIn'

export class SignIn extends Component {
  render() {
    const props = {
      ...this.props,
      submitHandler: submit
    }

    return (
      <SignInComponent {...props} />
    )
  }
}

export default reduxForm({
  form: 'signIn',
  fieldsForValidation: ['username', 'password']
})(SignIn)
