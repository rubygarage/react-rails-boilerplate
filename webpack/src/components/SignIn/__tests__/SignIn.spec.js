import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from 'components/SignIn';

describe('<SigniIn />', () => {
  let props = {
    handleSubmit: jest.fn(),
    submitHandler: jest.fn(),
    intl: {
      formatDate: jest.fn(),
      formatTime: jest.fn(),
      formatRelative: jest.fn(),
      formatNumber: jest.fn(),
      formatPlural: jest.fn(),
      formatMessage: jest.fn(),
      formatHTMLMessage: jest.fn(),
      now: jest.fn(),
    },
  };

  it('renders sign in form and submitting button is false', () => {
    props = {
      ...props,
      submitting: false,
    };
    const component = shallow(<SignIn {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders sign in form and pristine is true', () => {
    props = {
      ...props,
      pristine: true,
    };

    const component = shallow(<SignIn {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders sign in form and submitting is true', () => {
    props = {
      ...props,
      submitting: true,
    };

    const component = shallow(<SignIn {...props} />);

    expect(component).toMatchSnapshot();
  });
});
