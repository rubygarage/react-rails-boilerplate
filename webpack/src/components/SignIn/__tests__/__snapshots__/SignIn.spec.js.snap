// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigniIn /> renders sign in form and pristine is true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_in.login"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <div>
                <Field
                  className="mr-2"
                  component="input"
                  id="remember"
                  name="remember"
                  type="checkbox"
                />
                <label
                  htmlFor="remember"
                >
                  <FormattedMessage
                    id="sign_in.remember_me"
                    values={Object {}}
                  />
                </label>
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="sign_in.login"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_in.not_a_member_yet"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_up"
              >
                <FormattedMessage
                  id="sign_in.register"
                  values={Object {}}
                />
              </Link>
            </p>
            <p>
              <Link
                replace={false}
                to="/reset_password"
              >
                <FormattedMessage
                  id="sign_in.forgot_password"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SigniIn /> renders sign in form and submitting button is false 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_in.login"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <div>
                <Field
                  className="mr-2"
                  component="input"
                  id="remember"
                  name="remember"
                  type="checkbox"
                />
                <label
                  htmlFor="remember"
                >
                  <FormattedMessage
                    id="sign_in.remember_me"
                    values={Object {}}
                  />
                </label>
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={false}
                type="submit"
              >
                <FormattedMessage
                  id="sign_in.login"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_in.not_a_member_yet"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_up"
              >
                <FormattedMessage
                  id="sign_in.register"
                  values={Object {}}
                />
              </Link>
            </p>
            <p>
              <Link
                replace={false}
                to="/reset_password"
              >
                <FormattedMessage
                  id="sign_in.forgot_password"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SigniIn /> renders sign in form and submitting is true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_in.login"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <div>
                <Field
                  className="mr-2"
                  component="input"
                  id="remember"
                  name="remember"
                  type="checkbox"
                />
                <label
                  htmlFor="remember"
                >
                  <FormattedMessage
                    id="sign_in.remember_me"
                    values={Object {}}
                  />
                </label>
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="sign_in.login"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_in.not_a_member_yet"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_up"
              >
                <FormattedMessage
                  id="sign_in.register"
                  values={Object {}}
                />
              </Link>
            </p>
            <p>
              <Link
                replace={false}
                to="/reset_password"
              >
                <FormattedMessage
                  id="sign_in.forgot_password"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
